plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/MenuContributionInitializer.java

```diff
@@ -16,16 +16,20 @@
 
 package com.google.cloud.tools.eclipse.login.ui;
 
+import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
+import com.google.cloud.tools.eclipse.googleapis.internal.GoogleApiFactory;
 import org.eclipse.ui.IStartup;
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.commands.ICommandService;
 
 public class MenuContributionInitializer implements IStartup {
-
+  IGoogleApiFactory apiFactory = new GoogleApiFactory();
```

Comment:
Is this instance different than the apiFactory being passed down to other parts of the application? Are there going to be any issue with have multiple instances?

Reply:
I think we can fall back to a singleton as a follow up. Initially this class was supposed to take different credentials depending on which part of the plugin needed to access the apis. Now we have the same apis returned everywhere because it depends solely on the `AccountProvider` used by this class

Reply:
I added a follow up item in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/3712